packages/hebao_v1/contracts/stores/SecurityStore.sol

```diff
@@ -105,22 +154,40 @@ contract SecurityStore is DataStore
             );
             w.guardians.push(g);
             w.guardianIdx[guardian] = w.guardians.length;
+            w.guardianEffectiveTime[guardian] = effectiveTime;
         } else {
+            // you're not be able to add a pending removal guardian,
+            // use cancelGuardianRemoval instead.
+            require(
+                 w.guardianRemovalEffectiveTime[guardian] == 0 ||
+                 w.guardianRemovalEffectiveTime[guardian] >= now,
+                 "PENDING_REMOVAL_GUARDIAN"
+            );
+
             // Update the guardian
             w.guardians[pos-1].group = group;
+            w.guardianEffectiveTime[guardian] = effectiveTime;
+            delete w.guardianRemovalEffectiveTime[guardian];
```

Comment:
Seeing the code like this it seems like we can't allow updating the guardian because of security reasons:
- The attacker would currently would be able to disable the guardian immediately (because `guardianEffectiveTime` is set in the future) which could be dangerous.
- Even if `guardianEffectiveTime` is left unchanged (the previous behaviour) an attacker would be able to update the group number of all guardians immediately which he could use to his advantage.

So it seems like we should disable allowing the guardian group to be updated, otherwise we'd need a separate time limit for the group number so it can't be changed immediately.

Reply:
1. I think we should not allow a user to submit another removal when there is pending removal.
2. maybe we should not allow a user to change groups. all he can do is remove it and add a new one.

Reply:
1. Any reason in particular you wouldn't allow multiple removals? It will still take `pendingPeriod` to remove them all so the owner still has time to recover from this and cancel all the removals in that time period.
2. Yes, I think that would be best.

Reply:
Ok, then I will disable the guardian updating function.